src/FrozenClock.php

```diff
@@ -9,9 +9,9 @@ final class FrozenClock implements Clock
 {
     private DateTimeImmutable $now;
 
-    public function __construct(DateTimeImmutable $now)
+    public function __construct(?DateTimeImmutable $now = null)
```

Comment:
I see why you need this but what do you think introducing something like `SystemClock#freeze()`, which would return an instance of a FrozenClock (passing the current time as construction argument).

The main reason for this is to have a more explicit design and to not introduce a nullable parameter (which to me isn't always a good idea).

Reply:
Alternatively, we could introduce a named constructor instead of making this null (also handling the timezone to not blindly fallback to default value). Perhaps something like `fromNow()`?

I think I do prefer my previous suggestion, since it avoids code duplication. However, I'd like to hear your thoughts on the matter 😁